### PR TITLE
Pfappserver Updates

### LINKED ISSUE #3424: Can the unregdate field be searchable?
Use case is:

Our helpdesk is ran by students and I believe these helpdesk workers occasionally 'help out their bro' and either push the unregdate date out _far_ into the future or simply leave it blank so a device never unregisters.  This is fairly harmless, but still something I like to keep an eye on.  A couple of times a year I like to search for these oddities and fix them....and irritate the evil-doers!!  I currently search for this directly in the database.

Could the unregdate field searchable in the GUI....OR....put some tags on each Node screen showing the Admin user, action performed and time/date of the action?

Thanks!

### LINKED ISSUE #3431: Web admin Node search
It will be nice to have an advanced search on  “Detected Date,” “Registration Date,” and “Unregistration Date” since those are existing columns in the node page.

### PR DESCRIPTION
**Pull request by satkunas**
# Description
Continue work on the pfappserver

# Impacts
Increased PR count :100: 

# Issue
fixes #3431 
fixes #3424 

# Delete branch after merge
YES

### CODE REVIEW

**Comment by satkunas** [comment]
https://github.com/inverse-inc/packetfence/pull/3422/commits/546c1cb1f12de31177c77c3a4ed0ca64a20e70b0 fixes #3431 and #3424

File: html/pfappserver/root/static.alt/src/components/pfFormDatetime.vue [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+          handler: () => {
+            _this.validation.$touch()
+          },
+          time: this.debounce
+        })
+      }
+    },
+    toggle (event) {
+      event.preventDefault()
+      event.stopPropagation()
```
Comment (cgx) [inline html/pfappserver/root/static.alt/src/components/pfFormDatetime.vue]: Are those necessary? You already trigger this action with `@click.stop`.

File: html/pfappserver/root/static.alt/src/components/pfFormDatetime.vue [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+    labelCols () {
+      // do not reserve label column if no label
+      return (this.label) ? 3 : 0
+    },
+    datetimeConfig () {
+      const minMaxConfig = {
+        minDate: (this.min === '0000-00-00 00:00:00') ? new Date(-8640000000000000) : this.min,
+        maxDate: (this.max === '0000-00-00 00:00:00') ? new Date(8640000000000000) : this.max
+      }
+      return Object.assign(this.defaultConfig, minMaxConfig, this.config)
```
Comment (cgx) [inline html/pfappserver/root/static.alt/src/components/pfFormDatetime.vue:160]: Try using `Vue.set` instead and you should be able to remove your watchers on `min` and `max`.

File: html/pfappserver/root/static.alt/src/components/pfReportChart.vue [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+      } else {
+        // zip together then sort
+        let zip = values.map((e, i) => [e, labels[i]]).sort((a, b) => (a[0] === b[0]) ? 0 : (a[0] > b[0]) ? -1 : 1)
+        // truncate chart size limit
+        if (zip.length > this.chartSizeLimit) {
+          let other = zip.slice(this.chartSizeLimit).map(zip => zip[0])
+          zip = zip.slice(0, this.chartSizeLimit)
+          // push [sum(val), 'Other']
+          zip.push([other.reduce((sum, val) => sum + val), this.$i18n.t('Other')])
+          // "Paint It, Black" - Rolling Stones
```
Comment (cgx) [inline html/pfappserver/root/static.alt/src/components/pfReportChart.vue]: I prefer the later song title variation, without the comma ..

File: html/pfappserver/root/static.alt/src/components/pfReportChart.vue [hunk 4]
```diff
@@ -0,0 +1,243 @@
...
+  mounted () {
+    this.$nextTick(() => {
+      window.addEventListener('resize', this.getWindowWidth)
+      window.addEventListener('resize', this.getWindowHeight)
+      this.getWindowWidth()
+      this.getWindowHeight()
+    })
+  },
+  created () {
+  },
```
Comment (cgx) [inline html/pfappserver/root/static.alt/src/components/pfReportChart.vue]: Remove this empty definition.

File: html/pfappserver/root/static.alt/src/components/pfReportChart.vue [hunk 5]
```diff
@@ -0,0 +1,243 @@
...
+/**
+ * double-clicking legend causes user selection, disable
+ */
+.plotly * {
+  -webkit-touch-callout: none;
+  -webkit-user-select: none;
+  -khtml-user-select: none;
+  -moz-user-select: none;
+  -ms-user-select: none;
+  user-select: none;
```
Comment (cgx) [inline html/pfappserver/root/static.alt/src/components/pfReportChart.vue]: You should not have to add the browser-specific styles as they will be added by webpack/postcss.

File: html/pfappserver/root/static.alt/src/components/pfReportChart.vue [hunk 6]
```diff
@@ -0,0 +1,243 @@
...
+}
+/**
+ * Add btn-primary color(s) on hover
+ */
+.btn-group[rel=periodButtonGroup] button:hover {
+  color: #fff;
+  background-color: #007bff;
+  border-color: #007bff;
```
Comment (cgx) [inline html/pfappserver/root/static.alt/src/components/pfReportChart.vue]: Please use a Sass variable from Bootstrap instead.

File: html/pfappserver/root/static.alt/src/components/pfSearchBoolean.vue [hunk 7]
```diff
@@ -1,243 +1,267 @@
...
-  border-bottom-left-radius: $input-border-radius;
-}
-.sortable-drag .nodrag {
-  display: none;
-}
-</style>
+<template>
+  <b-container fluid class="px-0" v-if="!advancedMode">
+    <!-- BEGIN SIMPLE SEARCH -->
+    <b-container fluid class="rc px-1 py-1 bg-light">
```
Comment (cgx) [inline html/pfappserver/root/static.alt/src/components/pfSearchBoolean.vue]: You're rolling back my changes. I had voluntary remove the background colour to make the *simple search mode* visually similar to the *simple search mode with quick fields*. 

Also, please restore my changes in the SCSS.

File: html/pfappserver/root/static.alt/src/views/Nodes/index.vue [hunk 8]
```diff
@@ -122,6 +122,10 @@ export default {
   transform: rotate( -180deg );
   transition: transform 300ms ease;
 }
+.bd-sidenav .bd-toc-link:hover > svg {
+  color: #dc3545;
```
Comment (cgx) [inline html/pfappserver/root/static.alt/src/views/Nodes/index.vue]: Please use a Sass variable from Bootstrap instead.

File: html/pfappserver/root/static.alt/src/views/Reports/_components/ReportTable.vue [hunk 9]
```diff
@@ -33,47 +42,43 @@
           </b-container>
         </b-col>
       </b-row>
-      <b-table :items="items" :fields="visibleColumns" :sort-by="sortBy" :sort-desc="sortDesc" :sort-compare="sortCompare" v-model="tableValues"
-        @sort-changed="onSortingChanged" responsive hover></b-table>
+      <b-table stacked="sm" :items="items" :fields="visibleColumns" :per-page="pageSizeLimit" :current-page="requestPage" :sort-by="sortBy" :sort-desc="sortDesc" :sort-compare="sortCompare"
```
Comment (cgx) [inline html/pfappserver/root/static.alt/src/views/Reports/_components/ReportTable.vue]: You can't use both the `stacked` and `responsive` attributes at the same time. Keep `responsive`, remove `stacked`.

**Review by cgx — CHANGES_REQUESTED** [review]

File: html/pfappserver/root/static.alt/src/components/pfFormDatetime.vue (hunk 2 above)
Comment (satkunas) [reply html/pfappserver/root/static.alt/src/components/pfFormDatetime.vue:160]: min and max need to be watched, there's a 2 elements for a range:
`min_datetime`: sets the minimum for `max_datetime`
`max_datetime`: sets the maximum for `min_datetime`

This prevents our start date from exceeding our end date, and our end date from subceeding our start date.

**Review by cgx — APPROVED** [review]

**Pull request merged**